py-polars/tests/unit/interop/test_interop.py

```diff
@@ -812,13 +811,23 @@ def test_df_pycapsule_interface() -> None:
             "c": ["fooooooooooooooooooooo", "bar", "looooooooooooooooong string"],
         }
     )
-    out = pa.table(PyCapsuleStreamHolder(df))
+
+    capsule_df = PyCapsuleStreamHolder(df)
+    out = pa.table(capsule_df)
     assert df.shape == out.shape
     assert df.schema.names() == out.schema.names
 
-    df2 = pl.from_arrow(out)
-    assert isinstance(df2, pl.DataFrame)
-    assert df.equals(df2)
+    schema_overrides = {"a": pl.Int128}
+    expected_schema = pl.Schema([("a", pl.Int128), ("b", pl.String), ("c", pl.String)])
+
+    for arrow_obj in (
+        pl.from_arrow(capsule_df),  # capsule
+        out,  # table loaded from capsule
+    ):
+        df_res = pl.from_arrow(arrow_obj, schema_overrides=schema_overrides)
+        assert expected_schema == df_res.schema  # type: ignore[union-attr]
+        assert isinstance(df_res, pl.DataFrame)
+        assert df.equals(df_res)
```

Comment:
out of interest, any reason to use this instead of `assert_frame_equal(df, df_res, check_dtypes=False)`?

Reply:
None; just extended the existing test as-is 😂